src/graphic/gl/fields_to_draw.cc

```diff
@@ -111,8 +111,8 @@ void FieldsToDraw::reset(const Widelands::EditorGameBase& egbase,
 	assert(h_ > 0);
 
 	// Ensure that there is enough memory for the resize operation
-	size_t dimension = w_ * h_;
-	const size_t max_dimension = fields_.max_size();
+	uint32_t dimension = w_ * h_;
```

Comment:
this will not solve the problem of the multiplication overflowing before getting casted to the target type